README.md

```diff
@@ -64,42 +55,74 @@ batch_run_params = {'tasks': tasks,
 results = parallel_runner.run_batch(batch_run_params)
 ```
 
+KBParallel will give you back nested python dictionaries of results for every task that was run.
+Below is a description of this data structure.
 
-```
+```py
 # results data structure
 # ----------------------
 # The results of the function being called by kbparallel 
-# (align_reads_to_assembly_app), must be returned or it will not be accessable; 
+# (align_reads_to_assembly_app), must be returned or it will not be accessible; 
 # for instance, if align_reads_to_assembly_app creates a new alignment 
 # file, the path to this file must be returned in the output dictionary. 
 
-results =>
-{u'results': [{u'is_error': 0,
-               u'result_package': {u'error': None,
-                                   u'function': {u'function_name': u'align_reads_to_assembly_app',
-                                                 u'module_name': u'kb_Bowtie2',
-                                                 u'version': u'dev'
-                                                 },
-                                   u'result': [ ... Function call returned data ... ]
-                                   u'run_context': {u'job_id': u'81e3f2c4-5386-45f3-9bbf-d5a7bd23731a',
-                                                    u'location': u'local'}}
-              },
-              ...
-              ]
-
+{
+  'results': [
+    {
+      'is_error': 0,
+      'result_package': {
+        'error': None,
+        'function': {
+          'function_name': 'align_reads_to_assembly_app',
+          'module_name': 'kb_Bowtie2',
+          'version': 'dev'
+        },
+        'result': [ ... Method call return data ... ]
+        'run_context': {
+          'job_id': '81e3f2c4-5386-45f3-9bbf-d5a7bd23731a',
+          'location': 'local'
+        }
+      }
+    },
+    ...
+  ]
+}
 ```
 
 ### Some Examples
-I.
-For a simple hello world example that runs 3 tasks in parallel; each job creates a .txt file.  The jobs are run on 1 local & 2 njsw nodes:  
 
-to try it, search for __kbparallel example__ in 'dev'.  (also see https://gitlab.com/jfroula/kbparallel_example.git) .  
+For a simple hello world example that runs 3 tasks in parallel; each job creates a .txt file.  The jobs are run on 1 local & 2 njsw nodes. To try it, search for __kbparallel example__ in 'dev'.  (also see https://gitlab.com/jfroula/kbparallel_example.git) .  
+
+For an example that actually does something: search for __bowtie2__ or __Align Reads using Bowtie2 v2.3.2__.  (also see https://github.com/kbaseapps/kb_Bowtie2) .  
+
+This example is tricky because it calls the same function ("align" in Bowtie2Aligner.py) twice, once to set up the parallel tasks (runs this section first `if input_info['run_mode'] == 'sample_set'`) and then again to run each task (runs this section second `if input_info['run_mode'] == 'single_library':`).  This section actually does the work by calling single\_reads\_lib\_run.  
+
+## Development
```

Comment:
Added a section that should help guide others when they want to develop this codebase